client.go

```diff
@@ -62,7 +52,20 @@ type ClientInt interface {
 // information can be found in their respective documentation.
 // Numerous connection options may be specified by configuring a
 // and then supplying a ClientOptions type.
-type Client struct {
+
+type Client interface {
+	IsConnected() bool
+	Connect() Token
+	Disconnect(uint)
+	disconnect()
```

Comment:
That would be all that would need to be changed. I'm also not a maintainer of this project, so take my comment with a grain of salt. Thanks for the reply!